platform/linux-generic/odp_classification.c

```diff
@@ -1782,3 +1782,95 @@ uint64_t odp_pmr_to_u64(odp_pmr_t hdl)
 {
 	return _odp_pri(hdl);
 }
+
+static
+void print_cos_ident(struct cos_s *cos)
+{
+	ODP_PRINT("%s(%d)\n", cos->name, (int)cos->index);
+}
+
+static
+void print_queue_ident(odp_queue_t q)
+{
+	odp_queue_info_t info;
+
+	if (odp_queue_info(q, &info))
+		ODP_PRINT("%" PRIx64, odp_queue_to_u64(q));
+	else
+		ODP_PRINT("%s", info.name);
+
+	ODP_PRINT("\n");
+}
+
+static
+void print_hex(const void *vp, int len)
+{
+	const uint8_t *p = vp;
+
+	for (int i = 0; i < len; i++)
+		ODP_PRINT("%02x", (int)*p++ & 0xff);
+}
+
+static
+void cls_print_cos(struct cos_s *cos)
+{
+	print_cos_ident(cos);
+	ODP_PRINT("    queue: ");
+	print_queue_ident(cos->queue);
+
+	uint32_t num_rule = odp_atomic_load_u32(&cos->num_rule);
+	bool first = true;
```

Comment:
Usually, variables should be introduced in the beginning of a code block. E.g. a variable size array may be an exception to that (when size  is not know in the beginning of the block). Checkpatch does not seem to complain any more about this, but at least to me code is more readable when variables are introduced only on certain places (code block starts).

Reply:
I'll move these.

For the record, I disagree with this rule. Checkpatch doesn't require this, but it does require a blank line after declarations, which anyway makes it impossible to write naturally flowing code.